### PR TITLE
Make the Tags a set to avoid issues of DeDup

### PR DESCRIPTION
**Pull request by MushuEE**
Translated all tags into a set which should remove the need to ever track dupes and or sort by ID

Fix for #196

*I think there are easier ways to solve this but I am learning Go and wanted to use this as an opportunity to get a little better. Not hard feelings if this gets closed! Happy to solve the bug using `dedup` or another method. I think this is just more resilient long term. 

image

Tested :thumbsup:

### CODE REVIEW

**Comment by MushuEE** [comment]
@tstromberg I saw that you merged a quick fix for tags, would you like me to close this?

**Comment by tstromberg** [comment]
@MushuEE - Sorry about that. I merged a fix because I had made because I wanted to give more time to consider the implications of changing the data format. 

After looking more closely at the code, I like your PR better. Do you mind updating the merge conflict? I'll be able to merge it then.

**Comment by MushuEE** [comment]
Merge done 👍

File: site/collection.tmpl [hunk 1]
```diff
@@ -160,7 +160,7 @@
                 {{ end }}
               </td>
               <td class="cell-tags">
-                {{ range .Tags }}<div class="gh-tag tag-{{ .ID }}" title="{{ .Desc }}">{{ .ID }}</div> {{ end }}
+                {{ range $k, $_ := .Tags }}<div class="gh-tag tag-{{ $k.ID }}" title="{{ $k.Desc }}">{{ $k.ID }}</div> {{ end }}
```
Comment (tstromberg) [inline site/collection.tmpl:163]: Ahh, now I remember why I did this: consistent sorting of tags during display. I see that the Go developers did add sorting to template iteration though, so maybe this isn't a problem after all:

https://github.com/golang/go/commit/a5950df89e129e93f76aa8391f84beb460ef82be

If it is, we can always add a slice representation later. Thanks!

Comment (MushuEE) [reply site/collection.tmpl:163]: was a concern but yep, should be able to do one sort to slice right before template generation👍

**Pull request merged**